### PR TITLE
Haskell optional separate bin output

### PR DESCRIPTION
**Pull request by domenkozar**
The first commit adds most minimal separate output support (by not trigger recompilation on generic builder).

The second commit proves it's already useful.

### CODE REVIEW

**Comment by shlevy** [comment]
Is there a reason we wouldn't want this on by default?

**Review by shlevy — APPROVED** [review]

**Comment by domenkozar** [comment]
It's a huge rabbit hole to get all packages to build. That's why the PR was stuck for 2 years, let's ship infrastructure first and then do the default switch separately.

**Comment by domenkozar** [comment]
I'm going to merge this as it's a new toggle defaulting to old behaviour. Feel free to review it regardless and I'll address the comments.

**Comment by peti** [comment]
@domenkozar, it really sucks that you merge this kind of change without giving me (and others) any chance at all to review it first. This is quite anti-social behavior, IMHO, and I am very disappointed because you should know better.

**Comment by domenkozar** [comment]
@shlevy reviewed it?

**Comment by peti** [comment]
Very funny.

File: pkgs/development/haskell-modules/configuration-nix.nix [hunk 1]
```diff
@@ -94,6 +94,8 @@ self: super: builtins.intersectAttrs super {
   # Won't find it's header files without help.
   sfml-audio = appendConfigureFlag super.sfml-audio "--extra-include-dirs=${pkgs.openal}/include/AL";
 
+  cachix = enableSeparateBinOutput super.cachix;
```
Comment (puffnfresh) [inline pkgs/development/haskell-modules/configuration-nix.nix]: Think this should be in configuration-nix?

Comment (domenkozar) [reply pkgs/development/haskell-modules/configuration-nix.nix]: I'm not sure, could be also in `common`?

**Comment by Ericson2314** [comment]
https://github.com/input-output-hk/nix-tools building components separately makes me happy because then there are no `bin` or `lib` outputs to separate.

**Comment by domenkozar** [comment]
@Ericson2314 planning to switch to that soon but since this was a low hanging fruit it was really useful :)

**Pull request merged**